### ISSUE TITLE
Failure at IIS Install on VM

### ISSUE THREAD

**Issue by briankgarland** [gregcoffman/AzureVMSitecoreDevelopment#2]
See attached screenshot.
![IISError](https://user-images.githubusercontent.com/9375476/69923106-52186300-145f-11ea-9e2d-006f367f1174.jpg)

**Comment by gregcoffman** [comment]
Make sure your container is public, or that your VM can reach the files.  To validate, just navigate to one of your files from the VM.  e.g.  https://ContainerName.blob.core.windows.net/devvmfiles/InstallIisPsRemoting.ps1

Also, double-check the $BaseFileUri path in the Installer.ps1 matches where your files are located.

**Comment by briankgarland** [comment]
Thanks, that was it. I thought Azure blobs were public read access by default. Moving on....

**Comment by briankgarland** [comment]
Well, I got further but then this:

![image](https://user-images.githubusercontent.com/9375476/70072537-6b7ef380-15b4-11ea-8710-6dca4a82325b.png)

**Comment by briankgarland** [comment]
![image](https://user-images.githubusercontent.com/9375476/70072715-c6184f80-15b4-11ea-867b-21a6712c9a34.png)
![image](https://user-images.githubusercontent.com/9375476/70072744-cfa1b780-15b4-11ea-8ad3-b349b2288acb.png)
![image](https://user-images.githubusercontent.com/9375476/70072755-d7615c00-15b4-11ea-9e7b-b3f0975ac616.png)

**Comment by gregcoffman** [comment]
Check the event viewer on the VM and filter by _VmCustomScriptExtension_ and see if an error was thrown.  It may not like a parameter, or could have timed out due to the MSSQL download taking forever.

You're almost there.  I can also be reached on sitecorechat slack @gregcoffman.

**Comment by briankgarland** [comment]
No _VmCustomScriptExtension_ errors so maybe it is timing out on the MSSQL download. It that something that can be put in the blob container?

**Comment by gregcoffman** [comment]
You can absolutely do that.  It would still download the file from the blob, but I assume it would be faster than downloading it from Microsoft.

This was actually one of the enhancements I want to add is making this container private and allowing the VM to have access to it.  That way we're not publically hosting the software and we can put whatever we want in it.

I also noticed your license.xml file is in the blob storage, you'll want to remove that.  It gets copied from your local directly to the VM.

**Comment by briankgarland** [comment]
Yeah, I noticed the license file after I posted and promptly deleted it. So if I get the MSSQL installer into the blob I assume I just need to change your script where it points?
________________________________

**Comment by gregcoffman** [comment]
Correct.  Update the URL in  /AzureBlobFiles/InstallPrereqsSolrSql.ps1 and reupload it to the blob storage.

**Comment by briankgarland** [comment]
I moved everything into my blob container and changed all the pointers to it but it is still failing at this point:
![image](https://user-images.githubusercontent.com/9375476/70164541-92a1f780-167e-11ea-8cd8-8b07794e2977.png)

**Comment by gregcoffman** [comment]
To debug, navigate to the last numbered folder in C:\Packages\Plugins\Microsoft.Compute.CustomScriptExtension\ on the VM. In the **status** folder, you can find a log file the expanded powershell script that was executed.  Run it locally on the VM to see where it's failing.

**Comment by briankgarland** [comment]
Well, I still haven't figured it out. But I'm pretty sure it's failing at the point where it tries to download SQL Server Express: SQLEXPR_x64_ENU.exe. If I try downloading it manually from the VM, from either the Microsoft site or from my blob container, I get the following security warning message (probably because it's an .exe file). If I add either site to Trusted Sites, the download will work. So maybe another step in the script to do that before it attempts the download?
![image](https://user-images.githubusercontent.com/9375476/70460485-aa5cef80-1a73-11ea-9bc1-3c0b71f81991.png)
 
![image](https://user-images.githubusercontent.com/9375476/70460619-04f64b80-1a74-11ea-9c4b-833a7ef495f2.png)

**Comment by gregcoffman** [comment]
The IE error is likely unrelated to why the script is failing.   Please try the following:

Instead of remotely running the script as intended, just execute it directly on the VM so we can see the error.

Navigate to this folder in PowerShell on the VM: C:\Packages\Plugins\Microsoft.Compute.CustomScriptExtension\1.10.2\Downloads\

You'll find a bunch of numbered folders.  Choose the highest # and work your way backwards looking for the one containing InstallPrereqsSolrSql.ps1   In my case, it was folder 2.

Execute the file directly passing the variables:
InstallPrereqsSolrSql.ps1 -SqlAdminPassword YourPassword -VmDownloadPath C:\\VmSoftware\\

![image](https://user-images.githubusercontent.com/15002202/70494108-0d806d80-1ad8-11ea-96e7-92e336de67bd.png)

If that still isn't providing useful errors... another thing to try is to see if the executing script is somehow malformed.  Navigate to C:\Packages\Plugins\Microsoft.Compute.CustomScriptExtension\1.10.2\RuntimeSettings\  In each of these settings files, you'll find the PowerShell script it was trying to execute remotely.  

![image](https://user-images.githubusercontent.com/15002202/70494621-f80c4300-1ad9-11ea-8eb9-16bbf8263c36.png)

**Comment by briankgarland** [comment]
It ran without error on the VM using your instructions. The only difference I can see is that I passed in a SqlAdminPassword whereas in the install script I noticed that "sa" is used as the Sql admin account with no password. I went ahead and ran installer again from my local, using ".\Installation.ps1 @InstallParams -EnableCleanup -Verbose -SkipCreateVM -SkipPSRemoting" and it erred out like it did before. So, other than that Sql admin password, I don't know why it worked on the VM but fails from my local.
